salt/orchestrate/edx/deploy.sls

```diff
@@ -184,6 +184,15 @@ build_{{ purpose }}_{{ codename }}_edx_nodes:
           ansible_flags: "{{ ANSIBLE_FLAGS }}"
     {% endif %}
 
+{% if ENVIRONMENT == 'mitx-production' %}
+{# Recompile assets to pickup the production cloudfront domain #}
+compile_assets_for_edx_{{ purpose }}:
+  cmd.run:
+    - name: /edx/bin/edxapp-update-assets
+    - require:
+        - salt: build_{{ purpose }}_{{ codename }}_edx_nodes
+{% endif %}
```

Comment:
This should be in the `build_ami` state, not the deploy state. If we run it here then all of the instances will have a different content hash and the CDN won't work...